assets/js/modules/analytics/datastore/setup-flow.test.js

```diff
@@ -0,0 +1,231 @@
+	);
+};
+
+const populateAnalytics4DataStore = ( registry ) => {
+	registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetProperties( [
+		{
+			_id: '1000',
+			_accountID: '100',
+			name: 'properties/1000',
+			createTime: '2014-10-02T15:01:23Z',
+			updateTime: '2014-10-02T15:01:23Z',
+			parent: 'accounts/100',
+			displayName: 'Test GA4 Property',
+			industryCategory: 'TECHNOLOGY',
+			timeZone: 'America/Los_Angeles',
+			currencyCode: 'USD',
+			deleted: false,
+		},
+	],
+	{ accountID },
+	);
+	registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetWebDataStreams(
+		[
+			{
+				_id: '2000',
+				_propertyID: '1000',
+				name: 'properties/1000/webDataStreams/2000',
+				// eslint-disable-next-line sitekit/acronym-case
+				measurementId: '1A2BCD345E',
+				// eslint-disable-next-line sitekit/acronym-case
+				firebaseAppId: '',
+				createTime: '2014-10-02T15:01:23Z',
+				updateTime: '2014-10-02T15:01:23Z',
+				defaultUri: 'http://example.com',
+				displayName: 'Test GA4 WebDataStream',
+			},
+		],
+		{ propertyID: 'foobar' }
+	);
+};
+
+describe( 'modules/analytics setup-flow', () => {
+	let registry;
+
+	beforeAll( () => {
+		API.setUsingCache( false );
+	} );
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		// Receive empty settings to prevent unexpected fetch by resolver.
+		registry.dispatch( STORE_NAME ).receiveGetSettings( {} );
+	} );
+
+	afterAll( () => {
+		API.setUsingCache( true );
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getSetupFlowMode', () => {
+			it( 'returns “legacy” if the modules/analytics-4 store is not available ', async () => {
+				// just for this test create new registry with only this module
+				const newRegistry = createRegistry();
+
+				[
+					modulesAnalytics,
+				].forEach( ( { registerStore } ) => registerStore?.( newRegistry ) );
+
+				registry = newRegistry;
+				// Receive empty settings to prevent unexpected fetch by resolver.
+				registry.dispatch( STORE_NAME ).receiveGetSettings( {} );
+
+				expect( registry.select( STORE_NAME ).getSetupFlowMode() ).toBe( 'legacy' );
+			} );
+
+			it( 'should return “legacy” if isAdminAPIWorking() returns false ', () => {
+				registry.dispatch( MODULES_ANALYTICS_4 ).receiveError(
+					new Error( 'foo' ), 'getProperties', [ 'foo', 'bar' ]
+				);
+
+				expect( registry.select( MODULES_ANALYTICS_4 ).isAdminAPIWorking() ).toBe( false );
+
+				expect( registry.select( STORE_NAME ).getSetupFlowMode() ).toBe( 'legacy' );
+			} );
+
+			it( 'should return undefined if isAdminAPIWorking() returns undefined ', () => {
+				expect( registry.select( MODULES_ANALYTICS_4 ).isAdminAPIWorking() ).toBe( undefined );
+
+				expect( registry.select( STORE_NAME ).getSetupFlowMode() ).toBe( undefined );
+			} );
+
+			it( 'should return “ua” if there is no account selected', () => {
+				expect( registry.select( STORE_NAME ).getAccountID( accountID ) ).toBe( undefined );
```

Comment:
_"Actually, that's not true, because settings will return undefined when loading and when not set, argh!"_

Lol.  Right now it makes total sense because I've thought about it for a while

`undefined` means it hasn't loaded yet, but it can also mean it's not set.  Using `undefined` to mean loading is pretty interesting, but I suppose the downside is it means the absence of any value in JavaScript land (so it could be not set)